src/uwtools/config/support.py

```diff
@@ -15,7 +15,7 @@ def depth(d: dict) -> int:
     :param d: The dictionary whose depth to calculate.
     :return: The length of the longest path to a value in the dictionary.
     """
-    return (max(map(depth, d.values())) + 1) if isinstance(d, dict) else 0
+    return (max(map(depth, d.values())) + 1) if d and isinstance(d, dict) else 0
```

Comment:
Out of curiosity, what does this fix? An empty `dict` that we don't want to descend into?

Reply:
It looks like the `(max(map(depth, d.values()))` is returning an empty sequence. I haven't dug into exactly what situation in my YAML is causing this. I can if you think it's worth it.

Reply:
As long as `d` is valid before this, is it running in to an empty dict at the end of iterating through?

Reply:
Now I'm really curious. That expression should have an `int` value -- not a sequence/collection. I'm not sure why we have the parens around it, though. It would be great to know what is triggering bad behavior. A simple analog seems to do the right thing:
```
>>> def foo(b):
...   return (max([1, 3, 2]) + 1) if b else 0
... 
>>> foo(True)
4
>>> foo(False)
0
```
But it also seems fine without the parens:
```
>>> def bar(b):
...   return max([1, 3, 2]) + 1 if b else 0
... 
>>> bar(True)
4
>>> bar(False)
0
```

Reply:
If d is empty, then d.values() is an empty sequence. Now that Hera is back, I'm trying to dig a bit deeper into what exactly in my config is causing this.

Reply:
So, I finally found the culprit this morning...a SRW-prepared namelist has this in it, leading to the empty dict issue:

```
&nam_sfcperts
/

&nam_sppperts
/

&nam_stochy
/
```

Reply:
I get it now, it's essentially
```
>>> max(map(depth, {}.values()))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: max() arg is an empty sequence
```
Another way to solve it is to use the `default` keyword on `max()` to specify a value when the sequence is empty:
```
>>> max(map(depth, {}.values()), default=0)
0
```
I'm not sure if this would have made the potential problem clearer to me. Maybe I would have seen that the problem isn't recursion into an empty `dict` per se, but that `max` might be asked to operate on an empty iterable. I guess I'm always nervous about `if d` due to the many falsey values `d` could have that would make that expression `False`; when I see `if d` I have to think about the intent. Empty collection is one, but then I wonder about the others... I think using `default=0` in the call to `max()` avoids any nuance. FWIW, please do what you think best.

Reply:
I don't mind making a chance for readability/understanding. :)

Reply:
@maddenp-noaa Sorry it's taken a while to get back to it here. I saw you make this change in PR #392, so I'll merge develop in here to pick it up and re-test.